lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -356,37 +356,51 @@ export class UserAgentApplication {
       scopes = this.filterScopes(scopes);
     }
 
-    this._loginInProgress = true;
-    
-    this.authorityInstance.ResolveEndpointsAsync()
-      .then(() => {
-        const authenticationRequest = new AuthenticationRequestParameters(this.authorityInstance, this.clientId, scopes, ResponseTypes.id_token, this._redirectUri, this._state);
-        if (extraQueryParameters) {
-          authenticationRequest.extraQueryParameters = extraQueryParameters;
-        }
-
-        var loginStartPage = this._cacheStorage.getItem(Constants.angularLoginRequest);
-        if (!loginStartPage || loginStartPage === "") {
-            loginStartPage = window.location.href;
-        }
-        else {
-            this._cacheStorage.setItem(Constants.angularLoginRequest, "")
-        }
-
-        this._cacheStorage.setItem(Constants.loginRequest, loginStartPage, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.loginError, "");
-        this._cacheStorage.setItem(Constants.stateLogin, authenticationRequest.state, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.msalError, "");
-        this._cacheStorage.setItem(Constants.msalErrorDescription, "");
-        const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-        if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-          this._cacheStorage.setItem(authorityKey, this.authority);
-        }
-
-        const urlNavigate = authenticationRequest.createNavigateUrl(scopes)  + Constants.prompt_select_account + Constants.response_mode_fragment;
-        this.promptUser(urlNavigate);
-      });
+      this._loginInProgress = true;
+      const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
+      var idTokenObject;
+      if(adalIdToken) {
+           idTokenObject = Utils.extractIdToken(adalIdToken);
+      }
+      if (idTokenObject && idTokenObject.upn ) {
+          console.log("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
+          var extraQueryParam= "&" + Constants.login_hint+ "=" + idTokenObject.upn + "&" + Constants.domain_hint + "=" + Constants.organizations;
+          this.acquireTokenSilent([this.clientId], this.authority, this.getUser(), extraQueryParam)
+              .then((idToken) => {
+                  this._tokenReceivedCallback.call(this, "", idToken, "", "idToken", "");
```

Comment:
Will acquireTokenSilent automatically fallback to UI if silent refresh failed? If not, do we need to make sure here to fallback to interactive to not regress the previous functionality?

Reply:
In loginRedirect, if the user encounters an error due to passing of invalid scopes or if loginInProgress, we are passing the user defined state in the tokenReceivedCallback, But in the same loginRedirect, when you call ATS to achieve SSO, we should still pass the user defined state in the callback, instead to setting it to an empty string.

Reply:
In loginRedirect, we have a flag loginInProgress which gets set to true when we initiate login and then later on we set that to false after processing the redirect response. You will need to set this flag to false in the success as well as error case otherwise loginRedirect will not work for any subsequent calls.

Reply:
The same change for setting this flag to false has to be in loginPopup to avoid the error below:
if (this._loginInProgress) {
        reject(ErrorCodes.loginProgressError + Constants.resourceDelimeter + ErrorDescription.loginProgressError);
        return;
      }

Reply:
@rohitnarula7176 
1. Agree to pass the state in callback
2.Agree to set loginInProgress = false